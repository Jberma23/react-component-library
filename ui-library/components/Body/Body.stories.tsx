import { StoryFn, Meta } from "@storybook/react";
import { Body, BodyProps } from "./Body";

const sizes = ["small", "default", "large"] as BodyProps["size"][];

export default {
  title: "Library/Body",
  args: {
    bold: undefined,
    size: undefined,
    children: "Body text",
  },
  argTypes: {
    bold: {
      control: "boolean",
    },
    size: {
      options: sizes,
    },
  },
  parameters: {
    backgrounds: { default: "light" },
  },
} as Meta<typeof Body>;

export const Default: StoryFn<typeof Body> = (args) => <Body {...args} />;
