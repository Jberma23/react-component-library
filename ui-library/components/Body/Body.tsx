import { cx, VariantProps } from "cva";
import { HTMLAttributes } from "react";
import bodyStyles from "@ui-library/shared-styles/text-body";

export interface BodyProps<T extends React.ElementType = "div">
  extends VariantProps<typeof bodyStyles>,
    HTMLAttributes<HTMLParagraphElement> {
  as?: T;
  children: React.ReactNode;
}

export function Body<T extends React.ElementType = "div">({
  as,
  className,
  size,
  bold,
  ...rest
}: BodyProps<T>): JSX.Element {
  const Component = as || "div";
  return (
    <Component
      className={cx(bodyStyles({ bold, size }), className)}
      {...rest}
    />
  );
}
