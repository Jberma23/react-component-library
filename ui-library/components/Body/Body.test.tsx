import { render, screen } from "@testing-library/react";
import { Body } from "./Body";

// Since this is just a sample component, the tests below are not intended to be comprehensive
describe("ui-library/components/Body/Body.tsx", () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<Body>test body</Body>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", async () => {
    const { container } = render(
      <Body className={"test-class"}>test body</Body>
    );
    const bodyClass = screen.getByText("test body")?.getAttribute("class");
    expect(bodyClass).toContain("test-class");
    await expect(container).toBeA11y();
  });

  it("passes all a11y checks", async () => {
    const { container } = render(
      <Body className={"test-class"}>test body</Body>
    );
    await expect(container).toBeA11y();
  });
});
