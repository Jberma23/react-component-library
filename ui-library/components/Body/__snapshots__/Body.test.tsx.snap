// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-library/components/Body/Body.tsx renders with defaults 1`] = `
<DocumentFragment>
  <div
    class="leading-body font-regular text-body-regular"
  >
    test body
  </div>
</DocumentFragment>
`;
